### PR TITLE
Add Async, Guard & Form functionality

### PR DESCRIPTION
**Pull request by casiodk**

### CODE REVIEW

File: lib/hanikamu/concerns/form.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      end
+    end
+
+    included do
+      include ActiveModel::Validations
+    end
+
+    # this is not right
+    class_methods do
+      def call!(options = {})
```
Comment (casiodk) [inline lib/hanikamu/concerns/form.rb:25]: I want this method to first perform validations and then call super on the Class that it is included in

**Comment by alejanderl** [comment]
I like the ideas but I was thinking if we could keep the hanikamu service as a base library and make Async an extension in a new repo. 
I started something similiar with Hanikamu-Command

**Comment by alejanderl** [comment]
I think we'd discussed to implement the Hanikamu::Operation for the forms, etc? Seems like the right repo for this changes?

**Comment by casiodk** [comment]
> I think we'd discussed to implement the Hanikamu::Operation for the forms, etc? Seems like the right repo for this changes?

Ok, I will make a Hanikamu::Operation repo then and add all the extra functionality there then

**Pull request closed without merging**